### ISSUE TITLE
[prometheus-node-exporter] Incorrect url in chart index?

### ISSUE THREAD

**Issue by laithalissa** [prometheus-community/helm-charts#402]
Hi, this may not be a bug, but it's causing me problems so I thought I'd ask for clarification.

**Describe the bug**
Charts 1.11.2 and 1.12.0 have a different URL to what the README describes, as well as the older node-exporter charts.

The README says the repository is at https://prometheus-community.github.io/helm-charts

url for 1.12.0
https://github.com/prometheus-community/helm-charts/releases/download/prometheus-node-exporter-1.12.0/prometheus-node-exporter-1.12.0.tgz

url for 1.11.2
https://github.com/prometheus-community/helm-charts/releases/download/prometheus-node-exporter-1.11.2/prometheus-node-exporter-1.11.2.tgz

You can see this if you visit the index page https://prometheus-community.github.io/helm-charts/index.yaml and look at the most recent 2 versions of prometheus-node-exporter.

The urls for the other charts seem to be a mixture of the pages url and the release url.

**Version of Helm and Kubernetes**:

Helm Version:
n/a

Kubernetes Version:
n/a

**Which chart**:
prometheus-node-exporter
**Which version of the chart**:
1.11.2 and 1.12.0 

**What happened**:
I was having trouble setting up a proxy for this repo, and I discovered this might be the potential issue- my proxy modifies the index to update the urls, but it's failing on these two versions of prometheus

**What you expected to happen**:
I expect the URL to be https://prometheus-community.github.io/helm-charts/prometheus-node-exporter-1.12.0.tgz

**How to reproduce it** (as minimally and precisely as possible):
curl -s https://prometheus-community.github.io/helm-charts/index.yaml | grep 'prometheus-node-exporter'

**Changed values of values.yaml** (only put values which differ from the defaults):
n/a

**The helm command that you execute and failing/misfunctioning**:
n/a

**Helm values set after installation/upgrade:**
n/a

**Anything else we need to know**:

**Comment by torstenwalter** [comment]
Here a snippet from the curl command you describe:

```
  prometheus-node-exporter:
    name: prometheus-node-exporter
    - https://github.com/prometheus-community/helm-charts/releases/download/prometheus-node-exporter-1.12.0/prometheus-node-exporter-1.12.0.tgz
    name: prometheus-node-exporter
    - https://github.com/prometheus-community/helm-charts/releases/download/prometheus-node-exporter-1.11.2/prometheus-node-exporter-1.11.2.tgz
    name: prometheus-node-exporter
    - https://prometheus-community.github.io/helm-charts/prometheus-node-exporter-1.11.1.tgz
    name: prometheus-node-exporter
    - https://prometheus-community.github.io/helm-charts/prometheus-node-exporter-1.11.0.tgz
```

You are right that there is a mixture of locations. The reason is easy to explain: Since moving charts to this repository the [chart releaser action](https://github.com/helm/chart-releaser-action) was used to release charts. That one create a GitHub release for each new version and attaches the tgz files to it as can be seen in https://github.com/prometheus-community/helm-charts/releases/tag/prometheus-node-exporter-1.12.0
At the same time the [index.yaml](https://github.com/prometheus-community/helm-charts/blob/gh-pages/index.yaml) is updated. 
The `gh-pages` branch is used for the GitHub page which powers the repository.

Such an index.yaml is all it takes to have a helm chart repository and it's totally fine if it points to different locations.

If that does not work for you then try to update your helm version.

The tgz files, which are in this repository have been migrated from the old stable repository, which is now deprecated.

**Comment by laithalissa** [comment]
Thanks, that's what I suspected. FYI the github release url breaks artifactory virtual helm repos, as artifactory fails to rewrite the host in the index.yaml for those two versions of prometheus-node-exporter, I'll chase this up with their support.

**Comment by ruheenaansari34** [comment]
@laithalissa were you able to resolve your issue with artifactory. I'm facing the same issue and would like to know what worked for you.

**Comment by laithalissa** [comment]
@ruheenaansari34 
1. JFrog released an update (I think 6.23.7) which lets you set the repo prefix you're interested in, this is one possible solution, but as you might have guessed this will only work for one prefix, so it will simply change which repos work and which repos break. You might be able to hack this by adding multiple remote repos, each with a different base url, to make it work for each remote artifact you're interested in.
2. You can use JFrog's helm repo http://chartcenter.io/ as a workaround.
